### PR TITLE
chore: refactor `error` module

### LINKED ISSUE #3916: [Task]: Improve `error` module and general error handling
### Task description

The error handling could be improved a little bit further.

### Requirements

- [x] Separate console logging and storing error in `errorLog` store
- [x] Add new parameters: `localizeMessage`
- [x] Move `errorLog` to `error` module
- [x] Move `Error` type to `error` module and rename to `IError`
- [x] Cleanup existing code in `error` module
- [x] Add annotation comments
- [x] Move non-specific errors to corresponding module

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

The \`error\` module could be a little bit cleaner.

## Changelog

```
- Create "wallet" context module
- Move existing error logic to error module
- Add error parameters to log to console and save to error log
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Closes #3916 

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

Test instantiating an error using `BaseError`.

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/error/classes/base-error.class.ts [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
     constructor(params?: IErrorParameters) {
-        const message = params?.message ?? ''
+        const message = (params?.localizeMessage ? localize(params?.message) : params?.message) ?? ''
 
-        if (params?.logError) {
+        if (params?.logToConsole) {
             console.error(message)
         }
 
+        if (params?.saveToErrorLog) {
```
Comment (kraftjs) [inline packages/shared/lib/core/error/classes/base-error.class.ts:18]: Is this functionality that will be added in a future release?

Comment (maxwellmattryan) [reply packages/shared/lib/core/error/classes/base-error.class.ts:19]: No, you were just quick to review :sweat_smile:

**Comment by maxwellmattryan** [comment]
Possible improvement is to define a `UserError` and `AppError` which extend the `BaseError` class but simply use pre-defined constants to reduce parameter passing each time, e.g.

```typescript
const DEFAULT_USER_ERROR_PARAMETERS: Omit<IErrorParameters, 'message'> = {
    localizeMessage: true,
    logToConsole: true,
    saveToErrorLog: true,
    showNotification: true,
}

class UserError extends BaseError {
    constructor(params?: IErrorParameters) {
        super(
            ...DEFAULT_USER_ERROR_PARAMETERS,
            ...params,
        )
    }
}
```

**Comment by kraftjs** [comment]
I'm working with the base-error class right now and there is an issue when using the `saveToErrorLog` param. It only passes in the `message` param to the function `addError()` so the the displayed errors in the error log end up looking like:
image

They have no error type and the error time is invalid because it isn't passed.

Along these lines, I think there could be use for a generic `ErrorLogError` that takes an error as part of the constructor like:
```typescript
try () {
  // logic
} catch (err) {
  throw new ErrorLogError(err)
}
```

There are a lot of `console.error()` calls in the code and I think having a generic ErrorLogError (though I admit the name sucks) would help capture more of the user errors that weren't in the ErrorLog this beta test.

**Pull request merged**